generators/python-v2/codegen/src/ast/TypeInstantiation.ts

```diff
@@ -0,0 +1,179 @@
+import { assertNever } from "@fern-api/core-utils";
+import { AstNode } from "./core/AstNode";
+import { Writer } from "./core/Writer";
+import { Type } from "./Type";
+
+type InternalTypeInstantiation = Int | Float | Bool | Str | Bytes | List | Set | Tuple | Dict | None;
+
+interface Int {
+    type: "int";
+    value: number;
+}
+
+interface Float {
+    type: "float";
+    value: number;
+}
+
+interface Bool {
+    type: "bool";
+    value: boolean;
+}
+
+interface Str {
+    type: "str";
+    value: string;
+}
+
+interface Bytes {
+    type: "bytes";
+    value: string;
+}
+
+interface List {
+    type: "list";
+    values: TypeInstantiation[];
+}
+
+interface Set {
+    type: "set";
+    values: TypeInstantiation[];
+}
+
+interface Tuple {
+    type: "tuple";
+    values: TypeInstantiation[];
+}
+
+interface Dict {
+    type: "dict";
+    entries: DictEntry[];
+}
+
+interface DictEntry {
+    key: TypeInstantiation;
+    value: TypeInstantiation;
+}
+
+interface None {
+    type: "none";
+}
+
+export class TypeInstantiation extends AstNode {
+    private constructor(private readonly internalType: InternalTypeInstantiation) {
+        super();
+    }
+
+    public static int(value: number): TypeInstantiation {
+        return new TypeInstantiation({ type: "int", value });
```

Comment:
Mind using `new this(...)` like we do elsewhere (e.g. [Go](https://github.com/fern-api/fern/blob/3839a27d51fb4a3c6c597b520999fb1cdbf75297/generators/go-v2/codegen/src/ast/TypeInstantiation.ts#L181))?

Reply:
Yeah no problem